pallets/file-system/src/utils.rs

```diff
@@ -1556,6 +1564,10 @@ where
                             .bsps_volunteered
                             .saturating_sub(ReplicationTargetType::<T>::one());
 
+                        storage_request_metadata
+                            .data_server_sps
+                            .retain(|&x| x != bsp_id);
```

Comment:
I think we should instead remove this field entirely from the metadata and rely solely on `StorageRequestBsps` which holds all the confirmed BSPs of a storage request. Nodes can query this storage to find any potential confirmed BSPs and request the data. It would reduce complexity, additional storage and iteration as shown here. This was a premature addition to the metadata. 

https://github.com/Moonsong-Labs/storage-hub/blob/7c42d80fa25ac6e381b2a36f8597b96e75273571/pallets/file-system/src/lib.rs#L295-L303

Reply:
I agree with this and your proposed solution, as it's something I thought about when implementing the changes ("why do we have a variable in the request metadata for data servers when we have a separate storage for them"), so if @ffarall agrees I'll do the changes in this PR

Reply:
I agree with both.